osp/osp.c

```diff
@@ -564,6 +604,245 @@ osp_start_scan (osp_connection_t *connection, const char *target,
+ *
+ */
+static void
+target_append_as_xml (osp_target_t *target, GString *xml_string)
+{
+  xml_string_append (xml_string,
+                     "<target>"
+                     "<hosts>%s</hosts>"
+                     "<ports>%s</ports>",
+                     target->hosts ? target->hosts : "",
+                     target->ports ? target->ports : "");
+
+  if (target->credentials)
+    {
+      g_string_append (xml_string, "<credentials>");
+      g_slist_foreach (target->credentials,
+                       (GFunc) credential_append_as_xml,
+                       xml_string);
+      g_string_append (xml_string, "</credentials>");
+    }
+  xml_string_append (xml_string,
+                     "</target>");
+}
+
+/**
+ * @brief Append VT groups as XML to a string buffer.
+ *
+ * @param[in]     vt_group    VT group data.
+ * @param[in,out] xml_string  XML string buffer to append to.
+ */
+static void vt_group_append_as_xml (osp_vt_group_t *vt_group,
+                                    GString *xml_string)
+{
+  xml_string_append (xml_string,
+                     "<vt_group filter=\"%s\"/>",
+                     vt_group->filter);
+}
+
+/**
+ * @brief Append VT values as XML to a string buffer.
+ *
+ * @param[in]     id          Identifier of the vt_value.
+ * @param[in]     value       The value of the vt_value.
+ * @param[in,out] xml_string  XML string buffer to append to.
+ *
+ */
+static void
+vt_value_append_as_xml (gpointer id, gchar *value, GString *xml_string)
+{
+  xml_string_append (xml_string,
+                     "<vt_value id=\"%s\">%s</vt_value>",
+                     id ? id : "",
+                     value ? value : "");
+}
+
+/**
+ * @brief Append single VTs as XML to a string buffer.
+ *
+ * @param[in]     vt_single   Single VT data.
+ * @param[in,out] xml_string  XML string buffer to append to.
+ */
+static void vt_single_append_as_xml (osp_vt_single_t *vt_single,
+                                    GString *xml_string)
+{
+  xml_string_append (xml_string,
+                     "<vt_single id=\"%s\">",
+                     vt_single->vt_id);
+  g_hash_table_foreach (vt_single->vt_values,
+                        (GHFunc) vt_value_append_as_xml,
+                        xml_string);
+  xml_string_append (xml_string, "</vt_single>");
+}
+
+
+/**
+ * @brief Start an OSP scan against a target.
+ *
+ * @param[in]   connection      Connection to an OSP server.
+ * @param[in]   targets         Target hosts to scan.
+ * @param[in]   vt_groups       VT groups to use for the scan.
+ * @param[in]   vts             Single VTs to use for the scan.
+ * @param[in]   scanner_params  Table of scanner parameters.
+ * @param[in]   parallel        Number of parallel scans.
+ * @param[in]   scan_id         uuid to set for scan, null otherwise.
+ * @param[out]  error           Pointer to error, if any.
+ *
+ * @return 0 on success, -1 otherwise.
+ */
+int
+osp_start_scan_ext (osp_connection_t *connection, GSList *targets,
```

Comment:
How about using a osp_start_scan_opts_t argument instead of all these args?
That's the direction we started heading for new functions in gmp/gmp.c.

Reply:
Added the missing space in 6ad18f2060b0d9cb20d117605796a472a61179de

Reply:
I've changed the function in b1bd32c33138aa8aa329642e1a9f53871df2adf7